pkg/asset/cluster/tfvars.go

```diff
@@ -321,6 +321,12 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		for i, w := range workers {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*machinev1beta1.AzureMachineProviderSpec)
 		}
+		client := aztypes.NewClient(session)
+		hyperVGeneration, err := client.GetHyperVGenerationVersion(context.TODO(), masterConfigs[0].VMSize,
+			masterConfigs[0].OSDisk.ManagedDisk.StorageAccountType, masterConfigs[0].Location)
+		if err != nil {
+			return err
+		}
```

Comment:
This is clean, but I don't love making a second API call to retrieve the same information as was retrieved earlier.

I suppose simply checking whether the image in the master configs ends in `_v2` is too hacky.

AWS creates an install config metadata to hold this kind of information. Perhaps we could create a similar metadata for azure.

Reply:
> I suppose simply checking whether the image in the master configs ends in _v2 is too hacky.

If I recall, that is what the code was doing before that I commented was too fragile.

Reply:
> If I recall, that is what the code was doing before that I commented was too fragile.

Ah yes, I see that comment now and completely agree. The only suggestion I have for avoiding a second API call is to stick a variable in the Azure install config metadata:
https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/azure/metadata.go

I am happy with either solution.